MQTTClient-C/src/Network.h

```diff
@@ -0,0 +1,12 @@
+#ifndef MQTT_CLIENT_C_NETWORK_H
+#define MQTT_CLIENT_C_NETWORK_H
+
+int mqttread(void *network, unsigned char* read_buffer, int, int);
```

Comment:
I suggest to place forward declarations to `Network` structure here, to avoid using pointers to `void` everywhere.
Something like this:
`typedef struct Network Network;`

The same applies to `Timer` interface.